smarts/core/agent.py

```diff
@@ -92,10 +111,9 @@ class AgentSpec:
     """A callable to build an `smarts.core.agent.Agent` given `AgentSpec.agent_params` (default None)"""
     agent_params: Optional[Any] = None
     """Parameters to be given to `AgentSpec.agent_builder` (default None)"""
-
-    observation_adapter: Callable = lambda obs: obs
+    observation_adapter: Optional[Adapter] = None
     """An adaptor that allows shaping of the observations (default lambda obs: obs)"""
-    action_adapter: Callable = lambda act: act
+    action_adapter: Optional[Adapter] = None
```

Comment:
If `AgentSpec.xyz_adapter = None`, the following might fail and thus needs modification.

https://github.com/huawei-noah/SMARTS/blob/f59649c31e4f4a33fd77d3c39415e30270c6faea/smarts/env/hiway_env.py#L218-L220

Reply:
I have added a default adapter instead of None which will work for our existing examples